portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.data;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * A class for storing a Comment.
+ */
+public class Comment {
+    private String author;
+    private String subject;
+    private String text;
+
+    /**
+     * No argument constructor.
+     * Set fields with setter methods.
+     */
+    public Comment() {
+    }
+
+    public Comment(String author, String subject, String text) {
+        this.author = author;
+        this.subject = subject;
+        this.text = text;
+    }
+
+    public String getAuthor() {
+        return author;
+    }
+    
+    public void setAuthor(String author) {
+        this.author = author;
+    }
+
+    public String getSubject() {
+        return subject;
+    }
+
+    public void setSubject(String subject) {
+        this.subject = subject;
+    }
+
+    public String getText() {
+        return text;
+    }
+
+    public void setText(String text) {
+        this.text = text;
+    }
+
+    /**
+     * Returns a Comment from a request.
+     * Fills in appropriate defaults if attributes are null.
+     */
+    public static Comment makeComment(HttpServletRequest request) {
+        Comment c = new Comment();
+        c.setAuthor(request.getParameter("author"));
```

Comment:
Java suggestion:
 - Before setting the author, subject, and text fields of the comment, validate the input before creating the class. 
 - Example:

private static String getRequestParameterOrDefaultValue(HttpServletRequest request, String parameterName, String defaultValue) {
/**
 final String requestParameter = request.getParameter(parameterName);
 if (requestParameter is bad) {
  return defaultValue;
}
 return requestParameter.
*/
} 
  final String author = request.getParameter("author") != null

Reply:
Then use this function to fill in the comment.

Reply:
Thanks for the suggestion! Finished this change.